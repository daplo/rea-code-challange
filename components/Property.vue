<template>
	<b-card no-body class="overflow-hidden mb-2 p-0" :class="isSaved ? 'remove' : 'save'">
		<b-row no-gutters>
			<b-col md="12">
				<div class="agency-logo-wrapper" :style="branding">
					<img :src="item.agency.logo" class="agency-logo" />
				</div>
				<b-card-img
					:src="item.mainImage"
					alt="Image"
					class="rounded-0"
				></b-card-img>
			</b-col>
			<b-col md="12">
				<b-card-body>
					<b-card-text>
						<h3>{{ item.price }}</h3>
					
					</b-card-text>
				</b-card-body>

        <div class="buttonOverlay">
          	<b-button
              class="addOrRemoveButton"
              @click="removeOrSave"
              size="lg"
              :variant="isSaved ? 'warning' : 'primary'">
                {{ isSaved ? "Remove" : "Add" }}
            </b-button>
        </div>
			</b-col>
		</b-row>
	</b-card>
</template>

<script>
export default {
	name: "Property",
	props: ["item", "isSaved"],

	methods: {
		removeOrSave() {
			if (this.isSaved) {
				//if property is saved - move it to results
				this.$store.dispatch("moveToPropertyToResults", this.item);
			} else {
				//if property is in results - move it to saved
				this.$store.dispatch("moveToPropertyToSaved", this.item);
			}
		},
	},
	computed: {
		branding() {
			return {
				"background-color": this.item.agency.brandingColors.primary,
			};
		},
	},
};
</script>

<style lang="scss" scoped>
  .agency-logo {
    height: 40px;
    margin: auto;
  }

  .agency-logo-wrapper {
    padding: 2px;
    height: 44px;
    text-align: center;
  }

 .buttonOverlay {
    padding:1rem;
    position: absolute;
    z-index: 2;
    height: 10rem;
    left: 0;
    right:0;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    top:100%;
    bottom:-100%;
    z-index: 0;
    transition: .3s ease all;
    background: rgba(0,0,0,.6);

    .btn {
      min-width:180px
    }
  }


  .remove,
  .save {
    overflow: hidden;

    &:hover,
    &:focus {
      .buttonOverlay {
            top: unset;
            bottom:0;
      }
    }

  }
</style>
