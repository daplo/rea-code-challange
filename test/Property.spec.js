import { mount } from '@vue/test-utils'
import Property from '@/components/Property.vue'
import { BCard, BCardImg, BCardText, BCardBody,BButton, BCol, BRow } from 'bootstrap-vue'

describe('Property', () => {
  test('Property component', () => {
    const wrapper = mount(Property, {
        propsData: {
            isSaved: true,
            item: {
				price: "$726,500",
				agency: {
					brandingColors: {
						primary: "#00477F",
					},
					logo: "https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif",
				},
				id: "1",
				mainImage:
					"https://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg",
			},

        },

            stubs: {
                // used to register custom components
                'b-card':BCard,
                'b-card-img':BCardImg,
                'b-card-text':BCardText,
                'b-card-body':BCardBody,
                'b-button':BButton,
                'b-col':BCol,
                'b-row':BRow

        }
    })
    expect(wrapper.vm).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot();
  })
})
