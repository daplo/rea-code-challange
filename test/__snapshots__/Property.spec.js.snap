// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property Property component 1`] = `
"<div class=\\"card overflow-hidden mb-2 p-0 remove\\">
  <!---->
  <!---->
  <div class=\\"row no-gutters\\">
    <div class=\\"col-md-12\\">
      <div class=\\"agency-logo-wrapper\\" style=\\"background-color: rgb(0, 71, 127);\\"><img src=\\"https://i1.au.reastatic.net/170x32/d9e65c666e427e655fb63dcfe73f50d4ac7ff9a58c173db9474bd92e75b01070/main.gif\\" class=\\"agency-logo\\"></div> <img src=\\"https://i2.au.reastatic.net/640x480/20bfc8668a30e8cabf045a1cd54814a9042fc715a8be683ba196898333d68cec/main.jpg\\" alt=\\"Image\\" class=\\"rounded-0 card-img\\">
    </div>
    <div class=\\"col-md-12\\">
      <div class=\\"card-body\\">
        <!---->
        <!---->
        <p class=\\"card-text\\">
          <h3>$726,500</h3>
        </p>
      </div>
      <div class=\\"buttonOverlay\\"><button type=\\"button\\" class=\\"btn addOrRemoveButton btn-warning btn-lg\\">
          Remove
        </button></div>
    </div>
  </div>
  <!---->
  <!---->
</div>"
`;
